compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -630,6 +633,24 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         error: &SelectionError<'tcx>,
     ) {
         let tcx = self.tcx;
+
+        if tcx.sess.opts.unstable_opts.dump_solver_proof_tree == SolverProofTreeCondition::OnError {
+            self.infcx.probe(|_| {
+                let goal = Goal {
+                    predicate: root_obligation.predicate,
+                    param_env: root_obligation.param_env,
+                };
+                let tree = self
+                    .infcx
+                    .evaluate_root_goal(goal, GenerateProofTree::Yes(DisableGlobalCache::Yes))
+                    .1
+                    .expect("proof tree should have been generated");
+                let mut lock = std::io::stdout().lock();
+                let _ = lock.write_fmt(format_args!("{tree:?}"));
+                let _ = lock.flush();
```

Comment:
Why are you using println above but stdout::lock here?

Reply:
There's no `println` here

Reply:
i mean above in the pr, you're matching and using println